girder_worker_utils/tee.py

```diff
@@ -0,0 +1,71 @@
+import abc
+import sys
+
+import six
+
+
+def _set_stdout(stream):
+    """Set sys.stdout to a new file-like object.
+
+    This is a private function for re-assigning sys.stdout. It returns
+    the old value of sys.stdout.
+
+    """
+    old = sys.stdout
+    sys.stdout = stream
+    return old
+
+
+def _set_stderr(stream):
+    """Set sys.stderr to a new file-like object.
+
+    This is a private function for re-assigning sys.stderr. It returns
+    the old value of sys.stderr.
+
+    """
+    old = sys.stderr
+    sys.stderr = stream
+    return old
```

Comment:
For the above two functions, we could use python's handy swap semantics if we're inclined:

```
old, sys.stderr = sys.stderr, stream
return old
```